### DISCUSSION TITLE
Version 1.85 released

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#237] [General]
Available for testings in the `develop` branch, version `1.85` is about to be merged in `master`
Please reports any bug.

**Comment by inglor** [comment]
Do you mean 1.85 instead or 1.55?

**Reply by cyring** [reply]
> Do you mean 1.85 instead or 1.55?

Oh thanks for this. I'm fixing that right away.

**Comment by cyring** [comment]
**RC1** published in `develop`
# Driver
* Invoke the TDP-Config query from PCI/CSR registers for capable processors
* Store the Nominal Power settings accordingly to the Power Config Level
* Now query the SoC_SKL_VTD in Nominal mode [_no more experimental_]
# Daemon
* Don't substitute the Nominal-TDP with the Package-PL1
# UI/Client
* Print the IOMMU Version String [_Technologies window_]

# Use cases
## Nominal TDP
1. Level is default at `zero`
![2021-05-13-154108_644x452_scrot](https://user-images.githubusercontent.com/11563789/118133996-bc2d8980-b401-11eb-8e75-44efa0d591ba.png)
2. ... and Nominal TDP is `zero` (`Missing`)
![2021-05-13-154223_644x452_scrot](https://user-images.githubusercontent.com/11563789/118134199-f139dc00-b401-11eb-9a80-339a4b7ffb78.png)
3. Now we we change to level `1`
![2021-05-13-154434_644x452_scrot](https://user-images.githubusercontent.com/11563789/118134361-221a1100-b402-11eb-9c6c-dd530ddb5800.png)
4. ... and Nominal TDP is `35` watts
![2021-05-13-154534_644x452_scrot](https://user-images.githubusercontent.com/11563789/118134510-4bd33800-b402-11eb-90ff-0eec6b464c2d.png)
## IOMMU version
### Coffee Lake/HR
![2021-05-13-154655_644x452_scrot](https://user-images.githubusercontent.com/11563789/118134683-7d4c0380-b402-11eb-8aef-2e12cd67ece2.png)
### Zen2/Matisse
![2021-05-13-154749_644x550_scrot](https://user-images.githubusercontent.com/11563789/118134828-a2407680-b402-11eb-8d19-a291589d5f6d.png)
### Westmere/Gulftown
![2021-05-13-155003_644x452_scrot](https://user-images.githubusercontent.com/11563789/118135195-04997700-b403-11eb-9e3d-b71f86a320f3.png)

**Comment by cyring** [comment]
**RC2** published in develop
* Bit inversion fix in [en][dis]abling the Race To Halt Optimization
* "EEO_Disable" parameter and UI for Energy Efficiency Optimization
# For Intel SKL, KBL, CFL
_and probably, it may work with superior architectures_
* Observe the `Vcore` when toggling those optimizations
![2021-05-13-204245_644x452_scrot](https://user-images.githubusercontent.com/11563789/118172067-3c68e480-b42c-11eb-84ae-2a33f2caf681.png)

* Previous generations should be granted with such message:
![2021-05-13-204252_644x452_scrot](https://user-images.githubusercontent.com/11563789/118172222-6b7f5600-b42c-11eb-8204-3ae80e1edaab.png)

**Comment by cyring** [comment]
Version 1.85.5 **RC3**:
* Compute the TDP Limit Time Window when available in domain registers
![2021-05-14-030030_644x452_scrot](https://user-images.githubusercontent.com/11563789/118205450-a8fdd680-b460-11eb-8f2a-7b4c32f0658b.png)

**Comment by cyring** [comment]
tag [`1.85`](https://github.com/cyring/CoreFreq/releases/tag/1.85) is available

**Reply by inglor** [reply]
@cyring You released a hotfix for `1.85` version some days ago and you chose to reuse the same tag. This is a bad practice and breaks all efforts of packaging consistently. As a suggestion it would probably make sense to include on your tags the revision number, for example `1.85.6`, which you maintain in `coretypes.h`.

**Reply by cyring** [reply]
Hi @inglor 

I do agree with you. I did not want to impact your packages, thus I kept same rev number. It was a bad move.

Rather than GitHub, we need to find a better way to exchange b/c sometimes I can commit a lot of changes and I need _realtime_ answers, testing, and so on.

See my email in ` corefreq-cli -h`

Sincerely
Cyril
